lib/src/kvs/mem/mod.rs

```diff
@@ -101,6 +114,75 @@ impl Transaction {
 		// Return result
 		Ok(res)
 	}
+	/// Obtain a new change timestamp for a key
```

Comment:
`get_timestamp` and `set_versionstamped_key` still have a lot of code duplication across RocksDB, IndexedDB, InMem, and SpeeDB now.

However, as we already have full test coverage for each implementation, I think it's good to go for now and do a refactoring later(if necessary).

WDYT?

Reply:
Definitely good to go. This change is already quite big and we want to iterate, not land big changes IMO. We can refactor in the future. I have timestamp things I want to reconcile with this change as well :)